ci/unit/test_post_always.sh

```diff
@@ -13,17 +13,29 @@ else
   exit 1
 fi
 
-rm -rf run_test.sh vm_test
 DAOS_BASE="${SL_PREFIX%/install*}"
 NODE="${NODELIST%%,*}"
-
 mydir="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
 
+
+if [ $# -ge 1 ] && [ -n "$1" ]; then
+    if [ "$1" == "memcheck" ]; then
+        rm -rf run_test_memcheck.sh
+        WITH_VALGRIND=memcheck
+    fi
+else
+    rm -rf run_test.sh vm_test
```

Comment:
With the simplification, does this need to be moved into this conditional branch?